src/lib.rs

```diff
@@ -312,15 +322,15 @@ impl Trajectory for XTCTrajectory {
                 unsafe {
                     let path = path_to_cstring(&self.handle.path)?;
                     let path_p = path.into_raw();
-                    let code =
-                        xdrfile_xtc::read_xtc_natoms(path_p, &mut num_atoms as *const i32) as u32;
+                    let code = xdrfile_xtc::read_xtc_natoms(path_p, &mut num_atoms as *const i32);
```

Comment:
this cannot use `to_i32`?

Reply:
No, because the pointer is being cast rather than the type... but it turns out that all the raw pointer casts in lib.rs are unnecessary with a bit of borrow rejigging, which I've done.